apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -142,15 +146,36 @@ public void updateLastSequenceNumber(long lastSequenceNumber) {
    * @param apiVersion The version of the api used to get the message.
    * @param message The message that was being transformed when the error occurred.
    * @param exception The exception that was thrown while transforming the message.
-   * @throws IOException If there was an issue writing to the database.
+   * @throws IOException if there was an issue writing to the database.
+   * @throws ProcessingException If the error limit was reached.
    */
   @Override
   public void writeError(
       String apiVersion, TMessage message, DataTransformer.TransformationException exception)
-      throws IOException {
+      throws IOException, ProcessingException {
     entityManager.getTransaction().begin();
     entityManager.merge(createMessageError(apiVersion, message, exception.getErrors()));
     entityManager.getTransaction().commit();
+
+    checkErrorCount();
+  }
+
+  /**
+   * Checks if the allowed number of unresolved {@link MessageError}s in the database was exceeded.
+   *
+   * @throws ProcessingException If the error limit was reached.
+   */
+  @Override
+  public void checkErrorCount() throws ProcessingException {
+    var query =
+        entityManager.createQuery(
+            "select error from MessageError error where status = :status", MessageError.class);
+    query.setParameter("status", MessageError.Status.UNRESOLVED);
```

Comment:
curious...does every claim (or batch) processed make this query ?

Reply:
This method (and subsequent query) should only be executed at the start of each job, and when `writeError()` has been invoked, which only occurs when a transformation error happened.  In 99.99% of cases, we won't get any transformation errors, so the query will only get executed once when the pipeline job first starts.